sql-plugin/src/main/scala/com/nvidia/spark/rapids/VersionUtils.scala

```diff
@@ -43,4 +45,25 @@ object VersionUtils {
     val sparkFullVersion = ((sparkMajor.toLong * 1000) + sparkMinor) * 1000 + sparkBugfix
     sparkFullVersion.compareTo(fullVersion)
   }
+
+  /**
+   * Get the version used by JNI interface
+   * Must use `com.nvidia.spark.rapids.jni.Version` in the JNI interface
+   */
+  def getVersionForJni: VersionForJni = {
+    val sparkShimVersion = ShimLoader.getShimVersion
+    sparkShimVersion match {
+      case SparkShimVersion(a, b, c) =>
+        new VersionForJni(PlatformForJni.VANILLA_SPARK, a, b, c)
+      case DatabricksShimVersion(_, _, _, dbVer) =>
+        val major = dbVer.split("\\.")(0).toInt
+        val minor = dbVer.split("\\.")(1).toInt
+        new VersionForJni(PlatformForJni.DATABRICKS, major, minor, 0)
+      case ClouderaShimVersion(a, b, c, _) =>
+        new VersionForJni(PlatformForJni.CLOUDERA, a, b, c)
+      case other =>
+        // For other customer shim type, do not support
+        throw new IllegalStateException(s"Can not recognize shim version class: ${other.getClass}")
```

Comment:
After https://github.com/NVIDIA/spark-rapids-jni/pull/3434 is merged, will change to:

```scala
case unknown =>
  new VersionForJni(PlatformForJni.UNKNOWN, 0, 0, 0)
```